plutus-ledger-api/src/PlutusLedgerApi/V1/Interval.hs

```diff
@@ -79,14 +80,30 @@ instance Pretty a => Pretty (Extended a) where
     pretty PosInf     = pretty "+∞"
     pretty (Finite a) = pretty a
 
+-- See Note [Enumerable Intervals]
 -- | Whether a bound is inclusive or not.
 type Closure = Bool
 
 -- | The upper bound of an interval.
 data UpperBound a = UpperBound (Extended a) Closure
-    deriving stock (Haskell.Eq, Haskell.Ord, Haskell.Show, Generic)
+    deriving stock (Haskell.Show, Generic)
     deriving anyclass (NFData)
 
+-- | For an enumerable type, turn an upper bound into a single inclusive
+-- bounding value.
+--
+-- Since the type is enumerable, non-inclusive bounds are equivalent
+-- to inclusive bounds on the predecessor.
+--
+-- See Note [Enumerable Intervals]
+strictUpperValue :: Enum a => UpperBound a -> Extended a
```

Comment:
Haskell has `Haskell.Enum Double`, we don't, so I guess this is ok. But generally speaking I'm not sure whether `Enum` necessarily implies discreteness.

Reply:
> Haskell.Enum Double

Wow. Um, okay. Then I don't think we have a typeclass that expresses what we want. It's really something like `next (x) = inf { y | y > x }` or something.

Reply:
I updated the comments to try and express this issue.